src/test/java/ch/wisv/areafiftylan/products/service/TicketServiceTest.java

```diff
@@ -0,0 +1,664 @@
+    }
+
+    @Test
+    public void getAllTicketsOne() {
+        persistTicket();
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(1, tickets.size());
+    }
+
+    @Test
+    public void getAllTicketsTwo() {
+        persistTicket();
+        persistTicket();
+        Collection<Ticket> tickets = ticketService.getAllTickets();
+        assertEquals(2, tickets.size());
+    }
+
+    @Test
+    public void validateTicketFromInvalid() {
+        Ticket ticket = persistTicket();
+        ticket.setValid(false);
+        testEntityManager.persist(ticket);
+        assertEquals(false, ticket.isValid());
+        ticketService.validateTicket(ticket.getId());
+        assertEquals(true, ticket.isValid());
+    }
+
+    @Test
+    public void validateTicketAlreadyValid() {
+        Ticket ticket = persistTicket();
+        ticket.setValid(true);
+        testEntityManager.persist(ticket);
+        assertEquals(true, ticket.isValid());
+        ticketService.validateTicket(ticket.getId());
+        assertEquals(true, ticket.isValid());
+    }
+
+    @Test
+    public void validateTicketNotFound() {
+        Long id = 9999L;
+        thrown.expect(TicketNotFoundException.class);
+        thrown.expectMessage("Ticket not found");
+        ticketService.validateTicket(id);
+    }
+
+    @Test
+    public void validateTicketNull() {
+        thrown.expect(TicketNotFoundException.class);
+        thrown.expectMessage("Ticket not found");
+        ticketService.validateTicket(null);
+    }
+
+    @Test
+    public void requestTicketOfTypeStrings() {
+        User user = persistUser();
+        TicketType type = ticketTypeRepository.findByName(TEST_TICKET).get();
+        TicketOption pickupOption = ticketOptionRepository.findByName(PICKUP_SERVICE_OPTION).get();
+        TicketOption chMemberOption = ticketOptionRepository.findByName(CH_MEMBER_OPTION).get();
+        List<TicketOption> options = Arrays.asList(pickupOption, chMemberOption);
+        Set<TicketOption> optionSet = new HashSet<>(options);
+        String typeString = TEST_TICKET;
+        List<String> optionsString = Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION);
+
+        Ticket ticket = ticketService.requestTicketOfType(user, typeString, optionsString);
+
+        assertEquals(user, ticket.getOwner());
+        assertEquals(optionSet, ticket.getEnabledOptions());
+        assertEquals(type, ticket.getType());
+    }
+
+    @Test
+    public void requestTicketOfTypeStringsTypeNotAvailable() {
+        User user = persistUser();
+        String typeString = "unavailable_ticket";
+        List<String> optionsString = Arrays.asList(CH_MEMBER_OPTION, PICKUP_SERVICE_OPTION);
+
+        thrown.expect(TicketTypeNotFoundException.class);
+        thrown.expectMessage("TicketType ");
+
+        ticketService.requestTicketOfType(user, typeString, optionsString);
+    }
+
+    @Test
+    public void requestTicketOfTypeStringsTypeNull() {
```

Comment:
Does this test really differ from the previous one? What is your opinion on this @skamoen?